core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/Compression.java

```diff
@@ -147,25 +148,27 @@ public synchronized OutputStream createCompressionStream(OutputStream downStream
     },
 
     GZ(COMPRESSION_GZ) {
-      private transient DefaultCodec codec;
+      private transient AtomicReference<DefaultCodec> codec = new AtomicReference<DefaultCodec>();
 
       @Override
-      synchronized CompressionCodec getCodec() {
-        if (codec == null) {
-          codec = new DefaultCodec();
-          codec.setConf(conf);
+      CompressionCodec getCodec() {
+        DefaultCodec resultCodec = codec.get();
+        if (null == resultCodec) {
+          DefaultCodec newCodec = new DefaultCodec();
```

Comment:
Why not just share a singleton codec instance for each kind of codec? Looking at the implementations (Default, Gzip, Bzip2), they're all effectively stateless (sans the Configuration object you can set on them) because they're factories for streams.